content/blogs/2025-03-27-using-amazon-s3-tables-with-kestra.md

```diff
@@ -0,0 +1,307 @@
+      }
+
+  - id: s3_upload
+    type: io.kestra.plugin.aws.s3.Upload
+    accessKeyId: "{{ secret('AWS_ACCESS_KEY') }}"
+    secretKeyId: "{{ secret('AWS_SECRET_KEY') }}"
+    region: "{{ secret('AWS_REGION') }}"
+    from: "{{ outputs.ion_to_parquet.uri }}"
+    bucket: "s3-general-purpose-ecommerce"
+    key: "data/products.parquet"
+```
+
+### Creating S3 Table Bucket
+
+On the [S3 service](https://console.aws.amazon.com/s3/home) page on the AWS console, navigate to `Table buckets` from the left navigation menu.
+
+In case, you are opening the table buckets page for the first time, you will see a box at the top of the page about `Integration with AWS analytics services - New` with the `Enable integration` button. Click on the button and enable the integration of S3 table buckets with AWS analytics services like Amazon EMR, Amazon Redshift and Amazon Athena.
+
+![Enable Integration](/blogs/2025-03-27-using-amazon-s3-tables-with-kestra/enable_integration.png)
+
+Next, we will create a table bucket. Click on the `Create table bucket` button at the top of the page. On the `Create table bucket` page, provide an appropriate name for the table bucket, say `ecommerce-lakehouse`. Click on the `Create table bucket` button at the bottom of the page. This will create the new table bucket.
+
+![Create Table Bucket](/blogs/2025-03-27-using-amazon-s3-tables-with-kestra/create_table_bucket.png)
+
+### Providing IAM Access
+
+Navigate to [IAM service](https://console.aws.amazon.com/iam/home) on the AWS console. Navigate to `Roles` from the left navigation menu. Using the search box on the top of the page, ensure that you have `EMR_DefualtRole` and `EMR_EC2_DefaultRole` roles already present. If the roles are missing, you can create these default roles using `create-default-roles` as described [here](https://docs.aws.amazon.com/cli/latest/reference/emr/create-default-roles.html). Both these roles will be required for creating the EMR cluster.
+
+On the `Roles` page with IAM, search for `EMR_EC2_DefaultRole`, and click the same. On the `EMR_EC2_DefaultRole` role page, click on the `Add permissions` button, and from the dropdown that appears, click on `Attach Policies`. On the `Attach policy for EMR_EC2_DefaultRole` page, search for `AmazonS3TablesFullAccess`, select the `AmazonS3TablesFullAccess` policy and click on `Add permissions` button. This provides full access to S3 tables from the EC2 machines of the EMR.
+
+### Creating EC2 key pair
+
+Navigate to [EC2 service](https://console.aws.amazon.com/ec2/home) on the AWS console. From the left navigation menu, navigate to `Key pairs`. On the `Key pairs` page, click on the `Create key pair` button at the top of the page. On the `Create key pair` page, provide an appropriate name for the key pair, say `emr-ec2-key-pair`, and click on the `Create key pair` button at the bottom of the page. This will download the pem file associated with the key pair to your machine, and the new key pair will be created.
```

Comment:
```suggestion
Navigate to [EC2 service](https://console.aws.amazon.com/ec2/home) on the AWS console. From the left navigation menu, navigate to `Key pairs`. On the `Key pairs` page, select the `Create key pair` button at the top of the page. On the `Create key pair` page, provide an appropriate name for the key pair, say `emr-ec2-key-pair`, and select the `Create key pair` button at the bottom of the page. This will download the pem file associated with the key pair to your machine, and the new key pair will be created.
```